modules/gdscript/gdscript_editor.cpp

```diff
@@ -1186,14 +1260,21 @@ static void _find_identifiers_in_base(const GDScriptCompletionIdentifier &p_base
 
 	GDScriptParser::DataType base_type = p_base.type;
 
+	const HashMap<String, DocData::ClassDoc> class_doc_map = EditorHelp::get_doc_data()->class_list;
+
 	if (!p_types_only && base_type.is_meta_type && base_type.kind != GDScriptParser::DataType::BUILTIN && base_type.kind != GDScriptParser::DataType::ENUM) {
+		// Constructor for a class.
+		// 		var i = AnimatedSprite2D.n...  # <-- will suggest `new(...)`
 		ScriptLanguage::CodeCompletionOption option("new", ScriptLanguage::CODE_COMPLETION_KIND_FUNCTION, ScriptLanguage::LOCATION_LOCAL);
 		option.insert_text += "(";
 		option.display += U"(\u2026)";
 		r_result.insert(option.display, option);
 	}
 
 	while (!base_type.has_no_type()) {
+		// This HashMap is declared up here because it works across both
+		// the ENUM and BUILTIN cases.
+		HashMap<String, bool> enum_val_is_deprecated_map;
```

Comment:
For these I'd instead suggest using `HashSet<String>` and inserting if deprecated, and then using `has`, that avoids storing excessive entries

Reply:
Thanks for the suggestion, that makes a lot of sense! 😄